### PR TITLE
fix(acl): support all rules in JWT ACL

### PR DESCRIPTION
**Pull request by HJianBo**
### Backgourd
We had support JWT ACL in https://github.com/emqx/emqx/pull/7596. 
The docs https://www.emqx.io/docs/en/v5.0/security/authn/jwt.html#common-usage says it supported `all` rules, but the codes can't.

### Expected behaviour
After this change, we support it as the docs mentioned:
```
{
  "exp": 1654254601,
  "username": "myuser",
  "acl": {
    "pub": [...],
    "sub": [...],
    "all": [
      "testall1/${username}",
      "testall2/${clientid}",
      "testall3/#"
    ]
  }
}
```

I'm not sure what happened before, but I think supporting the `all` rule is more friendly.  I'll sync it to 5.0.x later if that makes sense to you guys.

### CODE REVIEW

**Review by savonarola — DISMISSED** [review]

File: apps/emqx_auth_jwt/src/emqx_auth_jwt.erl [hunk 1]
```diff
@@ -109,11 +109,17 @@ string_to_number(_) ->
...
-verify_acl(ClientInfo, #{<<"pub">> := PubTopics}, publish, Topic) when is_list(PubTopics) ->
-    verify_acl(ClientInfo, PubTopics, Topic);
-verify_acl(_ClientInfo, _Acl, _PubSub, _Topic) -> {stop, deny}.
+verify_acl(ClientInfo, Acl, PubSub, Topic) ->
+    Key = case PubSub of
+              subscribe -> <<"sub">>;
+              publish -> <<"pub">>
+          end,
+    case {maps:get(<<"all">>, Acl, []), maps:get(Key, Acl, [])} of
+        {Rules1, Rules2} when is_list(Rules1), is_list(Rules2) ->
```
Comment (zmstone) [inline apps/emqx_auth_jwt/src/emqx_auth_jwt.erl:118]:
```suggestion
        {Rules1, Rules2} when is_list(Rules1) orelse is_list(Rules2) ->
```

File: apps/emqx_auth_jwt/src/emqx_auth_jwt.erl [hunk 2]
```diff
@@ -109,11 +109,17 @@ string_to_number(_) ->
...
+            verify_acl(ClientInfo, Rules1 ++ Rules2, Topic);
+        {_, _} ->
```
Comment (zmstone) [inline apps/emqx_auth_jwt/src/emqx_auth_jwt.erl:120]: when does it end up here?

**Comment by zmstone** [comment]
please add jira link here.

File: apps/emqx_auth_jwt/src/emqx_auth_jwt.erl (hunk 2 above)
Comment (HJianBo) [reply apps/emqx_auth_jwt/src/emqx_auth_jwt.erl:120]: When the format of one of the `all` or `pub`, `sub` rules is incorrect? i.e:
```
{ "sub": ["t/a", "t/b"],
  "all": "bad_format_here"
}
```

**Review by savonarola — DISMISSED** [review]

File: apps/emqx_auth_jwt/src/emqx_auth_jwt.erl [hunk 3]
```diff
@@ -109,11 +109,20 @@ string_to_number(_) ->
...
+              subscribe -> <<"sub">>;
+              publish -> <<"pub">>
+          end,
+    Rules0 = lists:foldl(
+               fun(K, Acc) ->
+                       [case maps:get(K, Acl, undefined) of
+                            R when is_list(R) -> R;
+                            _ -> []
+                        end | Acc]
+               end, [], [<<"all">>, Key]),
```
Comment (zmstone) [inline apps/emqx_auth_jwt/src/emqx_auth_jwt.erl:123]:
```suggestion
    Rules0 = lists:map(
               fun(K, Acc) ->
                       case maps:get(K, Acl, undefined) of
                           R when is_list(R) -> R;
                           _ -> []
                       end
               end, [<<"all">>, Key]),
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**